x-pack/test/case_api_integration/security_and_spaces/tests/common/cases/tags/get_tags.ts

```diff
@@ -23,20 +34,160 @@ export default ({ getService }: FtrProviderContext): void => {
-
-      const { body } = await supertest
-        .get(CASE_TAGS_URL)
-        .set('kbn-xsrf', 'true')
-        .send()
-        .expect(200);
-
-      expect(body).to.eql(['defacement', 'unique']);
+      await createCase(supertest, getPostCaseRequest());
+      await createCase(supertest, getPostCaseRequest({ tags: ['unique'] }));
+
+      const tags = await getTags({ supertest });
+      expect(tags).to.eql(['defacement', 'unique']);
+    });
+
+    describe('rbac', () => {
+      it('User: security solution only - should read the correct tags', async () => {
+        await createCase(
+          supertestWithoutAuth,
+          getPostCaseRequest({ owner: 'securitySolutionFixture', tags: ['sec'] }),
+          200,
+          {
+            user: secOnly,
+            space: 'space1',
+          }
+        );
+
+        await createCase(
+          supertestWithoutAuth,
+          getPostCaseRequest({ owner: 'observabilityFixture', tags: ['obs'] }),
+          200,
+          {
+            user: obsOnly,
+            space: 'space1',
+          }
+        );
+
+        for (const scenario of [
+          {
+            user: globalRead,
+            expectedTags: ['sec', 'obs'],
+          },
+          {
+            user: superUser,
+            expectedTags: ['sec', 'obs'],
+          },
+          { user: secOnlyRead, expectedTags: ['sec'] },
+          { user: obsOnlyRead, expectedTags: ['obs'] },
+          {
+            user: obsSecRead,
+            expectedTags: ['sec', 'obs'],
+          },
+        ]) {
+          const tags = await getTags({
+            supertest: supertestWithoutAuth,
+            expectedHttpCode: 200,
+            auth: {
+              user: scenario.user,
+              space: 'space1',
+            },
+          });
+
+          expect(tags).to.eql(scenario.expectedTags);
+        }
+      });
+
+      for (const scenario of [
+        { user: noKibanaPrivileges, space: 'space1' },
+        { user: secOnly, space: 'space2' },
+      ]) {
+        it(`User ${scenario.user.username} with role(s) ${scenario.user.roles.join()} and space ${
+          scenario.space
+        } - should NOT get all tags`, async () => {
+          // super user creates a case at the appropriate space
+          await createCase(
+            supertestWithoutAuth,
+            getPostCaseRequest({ owner: 'securitySolutionFixture', tags: ['sec'] }),
+            200,
+            {
+              user: superUser,
+              space: scenario.space,
+            }
+          );
+
+          // user should not be able to get all tags at the appropriate space
+          await getTags({
+            supertest: supertestWithoutAuth,
+            expectedHttpCode: 403,
+            auth: { user: scenario.user, space: scenario.space },
+          });
+        });
+      }
+
+      it('should respect the owner filter when having permissions', async () => {
+        await Promise.all([
+          await createCase(
```

Comment:
I think we can remove the `await` next to the calls within the `Promise.all` right?